src/coreclr/src/vm/generics.cpp

```diff
@@ -488,6 +492,13 @@ ClassLoader::CreateTypeHandleForNonCanonicalGenericInstantiation(
         pInstDest[iArg] = inst[iArg];
     }
 
+    if (cbInstAndDict != 0 && pOldMT->GetClass()->GetDictionaryLayout() != NULL && pOldMT->GetClass()->GetDictionaryLayout()->GetMaxSlots() > 0)
```

Comment:
`cbInstAndDict` cannot be zero here.

Reply:
And `pOldMT->GetClass()->GetDictionaryLayout()` is likely too complex for the compiler to CSE so it would be nice to fetch it just once.

Reply:
Fixed